src/main/java/brandkon/Brands/BrandService.java

```diff
@@ -0,0 +1,43 @@
+package brandkon.Brands;
+
+import brandkon.Brands.DTO.BrandResponse;
+import brandkon.Brands.DTO.CreateBrand;
+import brandkon.categories.Category;
+import brandkon.categories.CategoryRepository;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+@Service
+public class BrandService {
+
+    private final BrandRepository brandRepository;
+    private final CategoryRepository categoryRepository;
+
+    public BrandService(BrandRepository brandRepository, CategoryRepository categoryRepository) {
+        this.brandRepository = brandRepository;
+        this.categoryRepository = categoryRepository;
+    }
+
+    public void Create(Long id, CreateBrand dto) {
+        Category category = categoryRepository.findById(id).orElseThrow();
+        Brand brand =
+                new Brand(dto.imageUrl(), dto.name(), category);
+        brandRepository.save(brand);
+    }
+
+//    public List<BrandResponse> findAll() {
+//        List<Brand> brands = brandRepository.findAll();
+//        return brands.stream().map(b -> new BrandResponse(
+//                b.getId(),b.getName(),b.getImageUrl())).toList();
+//    }
+
+    public List<BrandResponse> findByCategory(String slug) {
+        List<Brand> brands = brandRepository.findByCategorySlug(slug);
+        return brands.stream().map(b -> new BrandResponse(b.getId(), b.getName(), b.getImageUrl())).toList();
+    }
+
+    public BrandResponse findOne(Long brandId) {
+        Brand brand = brandRepository.findById(brandId).orElseThrow();
```

Comment:
```suggestion
        Brand brand = brandRepository.findById(brandId).orElseThrow(() -> new NoSuchElementException("존재하지 않는 브랜드 id: " + brandId));
```
더 구체적으로 작성하는 것이 더 좋아요